src/molecules/Modal/Modal.native.js

```diff
@@ -0,0 +1,246 @@
+  onSwipeComplete,
+  onBackButtonPress,
+}) => {
+  return type === 'centered' ? (
+    <Space margin={[3]}>
+      <RNModal
+        isVisible={visible}
+        onBackdropPress={onBackdropClick}
+        onBackButtonPress={onBackButtonPress}
+      >
+        <CenteredModalBody>
+          {onClose ? (
+            <Flex alignSelf="flex-end">
+              <Space padding={[0.5, 1, 0, 0]}>
+                <View>
+                  <Button
+                    testID="centered-cross"
+                    variant="tertiary"
+                    variantColor="tone" /**TODO: button uses shade of 800 but Figma is showing 980*/
+                    icon="close"
+                    size="medium"
+                    onClick={onClose}
+                  />
+                </View>
+              </Space>
+            </Flex>
+          ) : null}
+          {React.Children.map(children, (child) => {
+            return React.cloneElement(child, { ...child.props, type });
+          })}
+        </CenteredModalBody>
+      </RNModal>
+    </Space>
+  ) : type === 'fullscreen' ? (
+    <Space margin={[3, 0, 0, 0]}>
+      <RNModal isVisible={visible} hasBackdrop={false} onBackButtonPress={onBackButtonPress}>
+        <Flex flex={1}>
+          <FullscreenModalBody>
+            {React.Children.map(children, (child) => {
+              return React.cloneElement(child, { ...child.props, type, onClose });
+            })}
+          </FullscreenModalBody>
+        </Flex>
+      </RNModal>
+    </Space>
+  ) : type === 'bottomsheet' ? (
+    <Space margin={[0]}>
+      <Flex justifyContent="flex-end">
+        {/**TODO: confirm backdrop for bottomsheetModal */}
+        <RNModal
+          isVisible={visible}
+          swipeDirection={['down']}
+          onBackdropPress={onBackdropClick}
+          onSwipeComplete={onSwipeComplete}
+          onBackButtonPress={onBackButtonPress}
+        >
+          <BottomsheetModalBody>
+            {showDragBar ? (
+              <Flex alignItems="center">
+                <Space padding={[1, 0, 1.5, 0]}>
+                  <View>
+                    <Size height={0.5} width={8}>
+                      <FullScreenDragBar />
+                    </Size>
+                  </View>
+                </Space>
+              </Flex>
+            ) : null}
+            {React.Children.map(children, (child) => {
+              return React.cloneElement(child, { ...child.props, type });
+            })}
+          </BottomsheetModalBody>
+        </RNModal>
+      </Flex>
+    </Space>
+  ) : null;
+};
+
+Modal.Header = ModalHeader;
+Modal.Content = ModalContent;
+Modal.Footer = ModalFooter;
+
+Modal.propTypes = {
+  children: PropTypes.node.isRequired,
+  type: PropTypes.oneOf(['bottomsheet', 'centered', 'fullscreen']),
+  visible: PropTypes.bool,
+  onClose: PropTypes.func,
+  onBackdropClick: PropTypes.func,
+  onSwipeComplete: PropTypes.func,
+  onBackButtonPress: PropTypes.func,
+  showDragBar: PropTypes.bool,
```

Comment:
This shouldn't be a separate prop. How will you control 
```
showDragbar={true}
type="fullscreen"
```
dragbar is derived based on the type of Modal

Reply:
In the above case, the dragbar won't be shown. 

DragBar will only be shown in the Bottomsheet Modal. In Figma, they have Bottomsheet without dragBar.

Maybe, I should rename it to `showBottomsheetDragbar` to avoid confusion?

Reply:
Whutt. We'll ask them to have the dragbar in all the bottomsheet.